sdk/servicebus/azure-servicebus/tests/async_tests/test_queues_async.py

```diff
@@ -366,6 +368,59 @@ async def test_async_queue_by_queue_client_send_multiple_messages(self, uamqp_tr
             with pytest.raises(ValueError):
                 await receiver.peek_messages()
 
+            sender = sb_client.get_queue_sender(servicebus_queue.name)
+            receiver = sb_client.get_queue_receiver(servicebus_queue.name, max_wait_time=5)
+            async with sender, receiver:
+                # send previously unpicklable message
+                msg = {
+                    "body":"W1tdLCB7ImlucHV0X2lkIjogNH0sIHsiY2FsbGJhY2tzIjogbnVsbCwgImVycmJhY2tzIjogbnVsbCwgImNoYWluIjogbnVsbCwgImNob3JkIjogbnVsbH1d",
+                    "content-encoding":"utf-8",
+                    "content-type":"application/json",
+                    "headers":{
+                        "lang":"py",
+                        "task":"tasks.example_task",
+                        "id":"7c66557d-e4bc-437f-b021-b66dcc39dfdf",
+                        "shadow":None,
+                        "eta":"2021-10-07T02:30:23.764066+00:00",
+                        "expires":None,
+                        "group":None,
+                        "group_index":None,
+                        "retries":1,
+                        "timelimit":[
+                            None,
+                            None
+                        ],
+                        "root_id":"7c66557d-e4bc-437f-b021-b66dcc39dfdf",
+                        "parent_id":"7c66557d-e4bc-437f-b021-b66dcc39dfdf",
+                        "argsrepr":"()",
+                        "kwargsrepr":"{'input_id': 4}",
+                        "origin":"gen36@94713e01a9c0",
+                        "ignore_result":1,
+                        "x_correlator":"44a1978d-c869-4173-afe4-da741f0edfb9"
+                    },
+                    "properties":{
+                        "correlation_id":"7c66557d-e4bc-437f-b021-b66dcc39dfdf",
+                        "reply_to":"7b9a3672-2fed-3e9b-8bfd-23ae2397d9ad",
+                        "origin":"gen68@c33d4eef123a",
+                        "delivery_mode":2,
+                        "delivery_info":{
+                            "exchange":"",
+                            "routing_key":"celery_task_queue"
+                        },
+                        "priority":0,
+                        "body_encoding":"base64",
+                        "delivery_tag":"dc83ddb6-8cdc-4413-b88a-06c56cbde90d"
+                    }
+                }
+                await sender.send_messages(ServiceBusMessage(json.dumps(msg)))
+                messages = await receiver.receive_messages(max_wait_time=10, max_message_count=1)
+                if not uamqp_transport:
+                    pickled = pickle.loads(pickle.dumps(messages[0]))
+                    assert json.loads(str(pickled)) == json.loads(str(messages[0]))
+                else:
+                    with pytest.raises(TypeError):
+                        pickled = pickle.loads(pickle.dumps(messages[0]))
+                await receiver.complete_message(messages[0])
```

Comment:
What would happen if we tried to `complete` the message that had been pickled/unpickled?

Reply:
That would work - same as if we're completing "messages[0]".